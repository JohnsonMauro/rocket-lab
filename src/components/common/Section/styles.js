import styled from '@emotion/styled';

const SectionStyles = styled.section`
  margin-left: auto;
  margin-right: auto;
  padding: 0 40px;
  width: 100%;
`;

const SectionTitleHeader = styled.header`
  display: flex;
  margin-bottom: 24px;
`;

const SectionTitle = styled.h1`
  color: ${({ theme }) => theme.colors.blazeOrange};
  font-size: 22px;
  font-weight: 500;
  margin: 0;
`;

const SectionArticle = styled.article`
  display: flex;
  flex-direction: column;
`;

export { SectionStyles, SectionTitleHeader, SectionTitle, SectionArticle };
