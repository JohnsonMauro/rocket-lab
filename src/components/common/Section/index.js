import { SectionStyles, SectionTitleHeader, SectionTitle, SectionArticle } from './styles';

const Section = ({ children, title }) => (
  <SectionStyles>
    <SectionTitleHeader>
      <SectionTitle>{title}</SectionTitle>
    </SectionTitleHeader>
    <SectionArticle>{children}</SectionArticle>
  </SectionStyles>
);

export default Section;
