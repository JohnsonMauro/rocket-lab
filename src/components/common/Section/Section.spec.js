import React from 'react';
import { render } from '@testing-library/react';

import GlobalsContainer from '@test/GlobalsContainer';
import Section from './index';

describe('<Section />', () => {
  it('Renders the Section component', () => {
    const { container } = render(<Section title="Title Sample">Sample</Section>, {
      wrapper: GlobalsContainer,
    });
    expect(container).toMatchSnapshot();
  });
});
