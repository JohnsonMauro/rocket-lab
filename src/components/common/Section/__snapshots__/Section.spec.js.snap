// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Section /> Renders the Section component 1`] = `
<div>
  <section
    class="css-14vzaee-SectionStyles e18gs0ft0"
  >
    <header
      class="css-1pnqzzo-SectionTitleHeader e18gs0ft1"
    >
      <h1
        class="css-1y395ha-SectionTitle e18gs0ft2"
      >
        Title Sample
      </h1>
    </header>
    <article
      class="css-113vdcb-SectionArticle e18gs0ft3"
    >
      Sample
    </article>
  </section>
</div>
`;
